### DISCUSSION TITLE
disable async-to-functin-generator in configuration

### DISCUSSION THREAD

**Discussion by ShayRubach** [swc-project/swc#3296] [Q&A] [answered]
Can the [async to function](https://github.com/swc-project/swc/blob/681de03e4b0f6c28b173100c2255ce8f7d74d639/crates/swc_ecma_transforms_compat/src/es2017/async_to_generator.rs#L323) generation process can be disabled in configuration?

I wish to maintain
```javascript
(async () => {});
```
as it is, without having all of the 
```javascript
import regeneratorRuntime from "regenerator-runtime";
function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) {
   . . .
}

```
code added at all.
If not, is there other way to override it? I tried impl `fn async_to_generator() -> impl Fold + VisitMut {
` with an empty implementation in my .rs file but with no success (Rust beginner).

Thanks.

**Comment by kwonoj** [comment, marked as answer]
This is same as https://github.com/swc-project/swc/discussions/2788. We don't have explicit way to opt out specific transforms, in opposite as a workaround you can target higher level then add other transforms as needed.

**Reply by ShayRubach** [reply]
Hey and thanks for the quick reply. A few questions if I may: 😃 
By targeting a higher level you mean setting my `EsVersion` higher? (I'm using `EsVersion::Es2022`)
And with that being said, can you please direct me on how to configure plugins/transforms for swc? I'm not using any conf file at the moment, just straight up including the needed `swc` crates using Cargo and using:: them on my main.rs.

[Edit] Found the [.swcrc doc](https://swc.rs/docs/configuration/swcrc). Now I just need to figure out what to target and what transforms should I add instead, I guess?

**Comment by ShayRubach** [comment]
I think I solved it using your suggestion.

```rust
let _gen = comp.process_js(
        &handler, 
        _trns, 
        &Options{
            config: Config{
                jsc: JscConfig {
                    target: Some(EsVersion::Es2022),
                    ..Default::default()
                },
                minify: false,
                ..Default::default()
            },
            is_module: IsModule::Bool(true),
            ..Default::default()
        }
        ).unwrap();
```

My issue was that I had the 2022 target in the `parse_js` and not in the `process_js`.
Thanks!
